### ISSUE TITLE
NullPointerException customer statisfaction

### ISSUE THREAD

**Issue by coderDem** [IteraSpeed/OpenSpeedMonitor#10]
Hello,

I try to get the results for the customer statisfaction but all I get is this error message:
<code>
Executing action [showAll] of controller [de.iteratec.osm.result.EventResultDashboardController] caused exception Executing action [showAll] of controller [de.iteratec.osm.result.EventResultDashboardController] caused exception EventResultDashboardController has problem at line 157</code>

With this stack trace:
<code>
java.lang.NullPointerException

	at java.lang.Double.valueOf(Double.java:504)

	at de.iteratec.osm.result.ResultMeasuredValueService.getEventResultPropertyForCalculation(ResultMeasuredValueService.groovy:212)

	at de.iteratec.osm.result.EventResultDashboardService$_calculateResultMapForRawData_closure11_closure19.doCall(EventResultDashboardService.groovy:241)

	at de.iteratec.osm.result.EventResultDashboardService$_calculateResultMapForRawData_closure11.doCall(EventResultDashboardService.groovy:233)

	at de.iteratec.osm.result.EventResultDashboardService.calculateResultMapForRawData(EventResultDashboardService.groovy:229)

	at de.iteratec.osm.result.EventResultDashboardService$_calculateResultMap_closure8.doCall(EventResultDashboardService.groovy:214)

	at de.iteratec.osm.util.PerformanceLoggingService.logExecutionTime(PerformanceLoggingService.groovy:45)

	at de.iteratec.osm.result.EventResultDashboardService.calculateResultMap(EventResultDashboardService.groovy:212)

	at de.iteratec.osm.result.EventResultDashboardService.getEventResultDashboardHighchartGraphs(EventResultDashboardService.groovy:195)

	at de.iteratec.osm.result.EventResultDashboardController.fillWithMeasuredValueData(EventResultDashboardController.groovy:194)

	at de.iteratec.osm.result.EventResultDashboardController.showAll(EventResultDashboardController.groovy:157)

	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)

	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)

	at java.lang.Thread.run(Thread.java:745)</code>

What should I do to get it worked?

**Comment by nilskuhn** [comment]
Hi,

we fixed the null pointer february, 9th:
https://github.com/IteraSpeed/OpenSpeedMonitor/commit/6cebfc1b9eeebd4d1f09a20ea8bc97435dd47fef
With latest version you would just get an empty diagram cause no customer satisfactions got calcualated and persisted for your results.

For calculation of customer satisfactions there need to be a mapping _load time to percentage of customers that are satisfied by that load time_ in the database. By default there is no such mapping in a new OpenSpeedMonitor instance. For some customers we determined the mapping by an empirical study. 

We plan to add a way to add different mappings for the different pages you measure without the need for a survey or something (although that would result in a mapping that fits best your real customers expectations). Something like choosing from default profiles. In a simple form that could be linear functions: 
![csi_mapping_simple](https://cloud.githubusercontent.com/assets/1001134/7084144/4a90b972-df6a-11e4-9340-fdb6985bfde0.png)


At the moment you would have to manually fill db table customer_frustration with load times your customers got frustrated of waiting for the page to load.
